### PR TITLE
Added the new format for text memo in TransactionBuilder

### PR DESCRIPTION
**Pull request by ravitb**
Fixed the tests according to the new data.

### CODE REVIEW

File: scripts/src/blockchain/transactionBuilder.ts [hunk 1]
```diff
@@ -25,8 +34,8 @@ export class TransactionBuilder {
 		return this;
 	}
 
-	public addMemo(memo: string): this {
-		(this as any).memo = memo;
+	public addMemo(memo: Memo<MemoType.Text> | undefined, appId: string): this {
```
Comment (yosriz) [inline scripts/src/blockchain/transactionBuilder.ts]: the public API of memo shouldn't contain `appId` (developer shouldn't add this param, it will be propagated through the kin account it was created on).
Also, we should expose the entire options of `Memo` i.e. let the dev choose between id/hash/text memo (to provide a general builder), maybe add a dedicated for text memo that accepts `string`.

File: scripts/src/blockchain/transactionBuilder.ts [hunk 2]
```diff
@@ -25,8 +34,8 @@ export class TransactionBuilder {
...
+		(this as any)._transactionBuilder.memo = memo ? Memo.text('1-' + appId + '-' + memo.value) : Memo.none();
```
Comment (yosriz) [inline scripts/src/blockchain/transactionBuilder.ts]: no need for the cast (`this as any`)
can we const somehow the memo format?

File: scripts/src/blockchain/transactionBuilder.ts [hunk 3]
```diff
@@ -35,8 +44,12 @@ export class TransactionBuilder {
 		return this;
 	}
 
+	public set channel(channle: Channel | undefined) {
```
Comment (yosriz) [inline scripts/src/blockchain/transactionBuilder.ts]: any reason setter was added? the channel should be set on creation only (as it requires fetching sequence id from the network)

File: scripts/src/blockchain/transactionBuilder.ts [hunk 4]
```diff
@@ -35,8 +44,12 @@ export class TransactionBuilder {
...
+		(this as any)._transactionBuilder._channel = channle;
+	}
+
 	public get channel(): Channel | undefined {
-		return this._channel;
+		return (this as any)._transactionBuilder._channel;
```
Comment (yosriz) [inline scripts/src/blockchain/transactionBuilder.ts]: any reason for this change? the channel doesn't exist on the base transaction builder.

File: scripts/src/blockchain/transactionBuilder.ts (hunk 2 above)
Comment (yosriz) [inline scripts/src/blockchain/transactionBuilder.ts]: is the string concatenation safe in the case of an undefined memo?

File: tests/src/kinAccountTest.ts [hunk 5]
```diff
@@ -37,10 +37,10 @@ describe("KinAccount.createAccount", async () => {
 				memoText: memo
 			});
 		expect((txBuilder as any)._transactionBuilder.baseFee).toEqual(fee);
-		expect((txBuilder as any)._transactionBuilder.memo._value).toEqual(memo);
+		expect((txBuilder as any)._transactionBuilder.memo._value).toEqual('1-' + appId + '-' + memo);
```
Comment (yosriz) [inline tests/src/kinAccountTest.ts]: worth adding tests for empty and undefined memo (see comment on the potential undefined error at `TransactionBuilder`)

File: scripts/src/blockchain/transactionBuilder.ts (hunk 2 above)
Comment (ravitb) [reply scripts/src/blockchain/transactionBuilder.ts]: There is a check if the memo is undefined.
